.github/workflows/weekly-dependencies-pr.yml

```diff
@@ -0,0 +1,59 @@
+# https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions
+permissions:
+  actions: none
+  checks: none
+  contents: none
+  deployments: none
+  id-token: none
+  # This action can update/close issues
+  issues: write
+  discussions: none
+  packages: none
+  pages: none
+  pull-requests: write
+  repository-projects: none
+  security-events: none
+  statuses: none
+
+on:
+  push:
+    branches:
+    - weekly-dependency-updates
+    paths:
+    - 'Gemfile.lock'
+
+name: Weekly dependency PR workflow
+
+jobs:
+  create-pull-request:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Create Pull Request
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const { repo, owner } = context.repo;
+            const hasPR = await github.rest.pulls.list({
+              owner,
+              repo,
+              head: '${{ github.ref_name }}' 
```

Comment:
Can we hard code the branch name here and below? 👀 

```suggestion
              head: 'weekly-dependency-updates'
```

Reply:
I know it isn't an issue in this case as the branch list is locked down to `weekly-dependency-updates` and doesn't trigger on pull_requests, but I would like to avoid folk copy/pasting this pattern into other workflows without realizing the potential issue of using `github.ref_name` within scripts, as it can lead to code injection

i.e. If a workflow file triggered on pull request events:

```yml
# https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions
permissions:
  actions: none
  checks: none
  contents: none
  deployments: none
  id-token: none
  # This action can update/close issues
  issues: write
  discussions: none
  packages: none
  pages: none
  pull-requests: write
  repository-projects: none
  security-events: none
  statuses: none

on:
  pull_request:
    branches:
    - '*'
    paths:
    - 'Gemfile.lock'

name: Weekly dependency PR workflow

jobs:
  create-pull-request:
    runs-on: ubuntu-latest
    steps:
      - name: Create Pull Request
        uses: actions/github-script@v6
        with:
          script: |
            console.log('${{ github.ref_name }}');
```

A user could push a Github branch `foobar');console.log('helloworld123` and get code injection on CI:

<img width="1277" alt="image" src="https://user-images.githubusercontent.com/60357436/163716290-4f0ce6c6-e214-4064-a65f-1b742a19d9ca.png">

Again, not an issue in this case since we limit to branches `weekly-dependency-updates` and a normal user can't trigger push events, only `pull_request` events, but I'd just like to avoid using the pattern of `${{ github.ref_name }}` if possible 😄

Reply:
I see the concern here, on the other hand maybe it would be better to have an actions to detect and warn as apposed to ban usage of this refname.  This object is the official branch name which could significantly limit functionality for future actions.

Avoiding potential mis-use at the cost of valid use is sometimes a valid trade off so not apposed just offering another viewpoint.